client.go

```diff
@@ -369,15 +379,18 @@ func (c *Client) openConn(idx int, host string) (pconn *persistentConn, err erro
 
 	var conn net.Conn
 
+	ctx := context.Background()
+	if c.config.Timeout != 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, c.config.Timeout)
+		defer cancel()
+	}
+
+	pconn.debug("opening control connection to %s", host)
+	conn, err = c.config.DialContext(ctx, "tcp", host)
 	if c.config.TLSConfig != nil && c.config.TLSMode == TLSImplicit {
-		pconn.debug("opening TLS control connection to %s", host)
-		dialer := &net.Dialer{
-			Timeout: c.config.Timeout,
-		}
-		conn, err = tls.DialWithDialer(dialer, "tcp", host, pconn.config.TLSConfig)
-	} else {
-		pconn.debug("opening control connection to %s", host)
-		conn, err = net.DialTimeout("tcp", host, c.config.Timeout)
+		pconn.debug("configuring implicit TLS control connection to %s", host)
+		conn = tls.Client(conn, pconn.config.TLSConfig)
```

Comment:
Seems like you now need to check for err before using conn here.

Reply:
Yep.

Reply:
Fixed in ec080cc.